ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinUtils.kt

```diff
@@ -53,10 +50,19 @@ internal suspend fun OutgoingContent.toDataOrStream(): Any? {
                     var offset = 0
                     val read = channel.readAvailable(buffer, 0, 4096)
                     while (offset < read) {
+                        while (!outputStream.hasSpaceAvailable) {
+                            yield()
```

Comment:
How does it solve the problem?

Reply:
`outputStream.write()` will block if there is no space available and since it's not suspendable,  it will not throw on coroutine context cancellation. `yield()` adds a suspendable call, to make sure that we don't block